python/ql/lib/semmle/python/dataflow/new/internal/DataFlowDispatchPointsTo.qll

```diff
@@ -518,8 +562,34 @@ class SpecialCall extends DataFlowCall, TSpecialCall {
   }
 }
 
+/**
+ * A synthesized call inside a callable with a flow summary.
+ *
+ * For example, in
+ * ```python
+ * map(lambda x: x + 1, [1, 2, 3])
+ * ```
+ *
+ * there is a call to the lambda argument inside `map`.
+ */
+class SummaryCall extends DataFlowCall, TSummaryCall {
```

Comment:
In order for call-backs to work, you need to implement `lambdaCall` and `lambdaCreation` (see https://github.com/github/codeql/blob/main/ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll#L890-L931).

Reply:
`lambdaCall` is hiding [here](https://github.com/github/codeql/pull/8781/files#diff-3efed0c245564ccd514237561cd7e9f3a481afee65678c6be4518d98c25eaa75R537-R550). I did not find it at first, because the diff was collapsed, and had [a mild panic](https://github.com/github/codeql/pull/9099) while I thought we were looking at different versions of the code.

`lambdaCreation` is [here](https://github.com/github/codeql/pull/8781/files#diff-80716eaf55e2581b856950d844a39de74e7015eb05fb3fb56ea89662c13e56fdR927-R940).

Also, [the tests](https://github.com/github/codeql/pull/8781/files#diff-76ffbada01b4eb965e68a5348b591926230cf9090e6d1ccdf64935683e60763dR60-R61) show that call-backs do indeed work :-)